background/redux-slices/assets.ts

```diff
@@ -258,13 +258,30 @@ export const transferAsset = createBackgroundAsyncThunk(
           ERC20_INTERFACE,
           provider
         )
-        let transactionDetails = await token.populateTransaction.transfer(toAddress, assetAmount.amount)
-        toAddress = transactionDetails.to? transactionDetails.to : ""
-        data = transactionDetails.data? transactionDetails.data : ""
-        assetAmount = {
-          asset: assetAmount.asset,
-          amount: BigInt(0),
+        let fromShard = getShardFromAddress(fromAddress)
+        let toShard = getShardFromAddress(toAddress)
+        if(fromShard != toShard) { // ERC20x cross-chain transfer
+          let minerTip = BigInt(1000000000) * BigInt(calcEtxFeeMultiplier(fromShard, toShard))
+          let baseFee = BigInt(2000000000) * BigInt(calcEtxFeeMultiplier(fromShard, toShard))
+          let gasLimit = BigInt(100000)
+          let amount = (baseFee + minerTip) * gasLimit
+          let transactionDetails = await token.populateTransaction.crossChainTransfer(toAddress, assetAmount.amount, gasLimit, minerTip, baseFee)
+          toAddress = transactionDetails.to? transactionDetails.to : ""
+          data = transactionDetails.data? transactionDetails.data : ""
+          assetAmount = {
+            asset: QUAI,
+            amount: amount,
+          }
+        } else {
+          let transactionDetails = await token.populateTransaction.transfer(toAddress, assetAmount.amount)
+          toAddress = transactionDetails.to? transactionDetails.to : ""
+          data = transactionDetails.data? transactionDetails.data : ""
+          assetAmount = {
+            asset: QUAI,
+            amount: BigInt(0),
```

Comment:
why is the amount zero

Reply:
In a regular intra-shard ERC20 transfer, the 'amount' field is zero.